### PR TITLE
Tweak: Added Cross-Origin destination experiment to add `noopener` attribute on user-created links [ED-6581]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: includes/base/element-base.php [hunk 1]
```diff
@@ -395,6 +395,8 @@ public function add_link_attributes( $element, array $url_control, $overwrite =
 			$attributes = array_merge( $attributes, Utils::parse_custom_attributes( $url_control['custom_attributes'] ) );
 		}
 
+		$attributes = apply_filters( 'elementor/element/link-attributes', $attributes, $url_control );
```
Comment (Nevoss) [inline includes/base/element-base.php:398]: Seems like this is not the right place, this method is a helper method to add attributes to a link, I think it should be called when the link renders.

File: modules/cross-origin-destination/module.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			$rel_attributes = new Collection( preg_split(
+				'/\s+/',
+				$rel,
+				-1,
+				PREG_SPLIT_NO_EMPTY
+			) );
+
+			$rel_attributes->push( 'noopener' );
+
+			$attributes['rel'] = $rel_attributes->unique()->implode( ' ' );
```
Comment (Nevoss) [inline modules/cross-origin-destination/module.php:36]: Why not using, 
```php
$element->add_attribute('rel', $element->get_attribute('rel') . ' nooopener')
````
something like that?

File: modules/cross-origin-destination/module.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	protected function add_filters() {
+		add_filter( 'elementor/element/link-attributes', function ( $attributes ) {
+			$rel = empty( $attributes['rel'] ) ? '' : $attributes['rel'];
+
+			$rel_attributes = new Collection( preg_split(
+				'/\s+/',
+				$rel,
+				-1,
+				PREG_SPLIT_NO_EMPTY
```
Comment (Nevoss) [inline modules/cross-origin-destination/module.php:31]: Can you do some normal split on space? (like normal people 😉 😆 )

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/cross-origin-destination/module.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+			$rel_attributes->push( 'noopener' );
+
+			$attributes['rel'] = $rel_attributes->unique()->implode( ' ' );
+
+			return $attributes;
+		} );
+	}
+
+	public static function get_experimental_data() {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply modules/cross-origin-destination/module.php:42]: Done

File: modules/cross-origin-destination/module.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		return [
+			'name' => 'cross-origin-destinations',
+			'title' => esc_html__( 'Cross-Origin Destination', 'elementor' ),
+			'description' => esc_html__( 'This feature adds rel="noopener" to all external links. It prevents a new page from being able to access the window.opener property and ensures it runs in a separate process.', 'elementor' ),
+			'release_status' => Experiments_Manager::RELEASE_STATUS_BETA,
+			'default' => Experiments_Manager::STATE_INACTIVE,
+			'new_site' => [
+				'default_active' => true,
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply modules/cross-origin-destination/module.php:50]: Done

File: modules/cross-origin-destination/module.php (hunk 3 above)
Comment (StyleShit) [reply modules/cross-origin-destination/module.php:31]: Done

**Review by Nevoss — APPROVED** [review]

File: modules/cross-origin-destination/module.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+				$rel,
+				-1, // Default value.
+				PREG_SPLIT_NO_EMPTY
+			);
+
+			$rel = new Collection( $rel );
+
+			$rel->push( 'noopener' );
+
+			$attributes['rel'] = $rel->unique()->implode( ' ' );
```
Comment (StyleShit) [inline modules/cross-origin-destination/module.php:54]:
```suggestion
			$attributes['rel'] = ( new Collection( $rel ) )
				->push( 'noopener' )
				->unique()
				->implode( ' ' );
```

**Pull request closed without merging**